### PR TITLE
Added import functionality for epgp

### PR DESCRIPTION
**Pull request by elliotmendiola**
Updated parseImportData in Import.lua to function sans dkp and with spent / points as well as track action count properly and update spent in the standings
Updated LedgerEntries.lua to function with Set command for spent points in epgp
Updated PointManager.lua to pass spent property into set command and added set spent mutator

### CODE REVIEW

**Comment by elliotmendiola** [comment]
image
image
image
I checked the numbers for accuracy

File: Integrations/Import.lua [hunk 1]
```diff
@@ -61,17 +63,34 @@ local function ParseImportData(self)
                 }
             end
             rosterMap[roster.uid].players[#rosterMap[roster.uid].players+1] = player.guid
-            if player.dkp ~= 0 then
+
+            if player.dkp and player.dkp ~= 0 then
```
Comment (lantisnt) [inline Integrations/Import.lua:67]: If we are moving from `dkp` nomenclature to `points` this one is redundant and shouldn't be used here as `points` would cover DKP

File: Migration.lua [hunk 2]
```diff
@@ -85,9 +86,16 @@ function Migration:MigrateCommunityDKP()
...
+function Migration:MigrateBastion()
+    LOG:Trace("Migration:MigrateBastion()");
+    self:_MigrateBastion();
+end
+
+local function NewRoster(name, epgp)
     local timestamp = Migration:GetOldTimestampUnique()
-    local roster = LEDGER_ROSTER.Create:new(timestamp, name, CONSTANTS.POINT_TYPE.DKP)
+    local pointType = CONSTANTS.POINT_TYPE.DKP;
+    if (epgp) then pointType = CONSTANTS.POINT_TYPE.EPGP end
```
Comment (lantisnt) [inline Migration.lua:97]:
```suggestion
    local pointType
    if epgp then
        pointType = CONSTANTS.POINT_TYPE.EPGP
    else
        pointType = CONSTANTS.POINT_TYPE.DKP
    end
```

File: Migration.lua [hunk 3]
```diff
@@ -128,6 +136,21 @@ local function UpdatePoints(uid, targets, value)
     entry:setTime(timestamp)
     entry:setCounter(timestampCounter[timestamp])
     LedgerManager:Submit(entry)
+
+    if (spent ~= nil) then
```
Comment (lantisnt) [inline Migration.lua:140]: that's doubling the code a lot. I'd rather see the whole `UpdatePoints` called twice in the migration method, with new parameter `UpdatePoints(uid, targets, value, isSpent)` that will propagate to the ledger entry creation.

File: Migration.lua [hunk 4]
```diff
@@ -360,12 +383,58 @@ function Migration:_MigrateCommunity()
     LOG:Message(CLM.L["Import complete"])
 end
 
+function Migration:_MigrateBastion()
```
Comment (lantisnt) [inline Migration.lua:386]: I like this function, and I see good potential in reusing it for CEPGP. Could you make it more generic and then parametrize it for `Bastion`? it seems it's basically changing officer note format for the addons.

**Review by lantisnt — CHANGES_REQUESTED** [review]
In general it looks good. Just few refinements + please remove trailing `;` as those are not used in lua unless you wan't to execute multiple actions in same line of code

File: Integrations/Import.lua (hunk 1 above)
Comment (elliotmendiola) [reply Integrations/Import.lua:67]: I wasn't able to test the export functionality with a dkp setup, I agree with the sentiment but was unable to confirm the ability to remove this field.  Additionally this version maintains compatibility with old exports while only including an additional null check

File: Migration.lua (hunk 3 above)
Comment (elliotmendiola) [reply Migration.lua:144]: Fair enough, will update

**Review by lantisnt — APPROVED** [review]

**Pull request merged**